File: SBL/Java/Parser.java
package fh.scheme.parser;

import java.io.IOException;

public class Parser {
   private Lexer input;
   protected Token lookahead;
   
   public Parser(Lexer input) {
	   this.input = input;
	   consume();
   }
   
   public void consume() {
	   try {
		lookahead = input.nextToken();
	} catch (IOException e) {
		e.printStackTrace();
	}
   }
   
   public void match(int type) {
	   if(lookahead.getType() == type) {
		   consume();
	   } else throw new RuntimeException("parsing error: expecting " + input.getTokenName(type) 
	   + "; found " + lookahead);
   }
}

File: SBL/Parser.cs
using System;

public class Parser
{
    private Lexer input;
    protected Token lookahead;

    public Parser(Lexer input)
    {
        this.input = input;
        Consume();
    }

    public void Consume()
    {
        try
        {
            lookahead = input.nextToken();
        }catch(Exception e)
        {
            Console.Write(e.StackTrace);
        }
    }

    public void Match(int type)
    {
        if (lookahead.getType() == type)
        {
            Consume();
        }
        else throw new ParserException("Expecting " + input.getTokenName(type)
                                        + "; found " + lookahead);
    }
}